prog/prio.go

```diff
@@ -86,33 +80,33 @@ func (target *Target) calcResourceUsage() map[string]map[int]float32 {
 						if i < len(a.Desc.Kind)-1 {
 							w = 0.2
 						}
-						noteUsage(uses, c, float32(w), str)
+						noteUsage(uses, c, float32(w), a.Dir(), str)
 					}
 				}
 			case *PtrType:
 				if _, ok := a.Type.(*StructType); ok {
-					noteUsage(uses, c, 1.0, "ptrto-%v", a.Type.Name())
+					noteUsage(uses, c, 1.0, a.Dir(), "ptrto-%v", a.Type.Name())
 				}
 				if _, ok := a.Type.(*UnionType); ok {
-					noteUsage(uses, c, 1.0, "ptrto-%v", a.Type.Name())
+					noteUsage(uses, c, 1.0, a.Dir(), "ptrto-%v", a.Type.Name())
 				}
 				if arr, ok := a.Type.(*ArrayType); ok {
-					noteUsage(uses, c, 1.0, "ptrto-%v", arr.Type.Name())
+					noteUsage(uses, c, 1.0, a.Dir(), "ptrto-%v", arr.Type.Name())
 				}
 			case *BufferType:
 				switch a.Kind {
 				case BufferBlobRand, BufferBlobRange, BufferText:
 				case BufferString:
 					if a.SubKind != "" {
-						noteUsage(uses, c, 0.2, fmt.Sprintf("str-%v", a.SubKind))
+						noteUsage(uses, c, 0.2, a.Dir(), fmt.Sprintf("str-%v", a.SubKind))
 					}
 				case BufferFilename:
-					noteUsage(uses, c, 1.0, "filename")
+					noteUsage(uses, c, 1.0, DirIn, "filename")
```

Comment:
We use DirIn here. Please think if we need to use DirIn instead of a.Dir() in other (non-resource) cases.
We only pass resource value from one call to another, but for the other things we don't pass anything. For other things noteUsage simply means "there is some relation between these calls".